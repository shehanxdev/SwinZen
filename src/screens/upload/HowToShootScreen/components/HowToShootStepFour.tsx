import { View } from 'react-native';
import FastImage from 'react-native-fast-image';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import {
  Color,
  FaceOnGIF,
  HOW_TO_SHOOT_GIF_ASPECT_RATIO,
  LandscapeGIF,
  TextAlignment,
  TextVariant,
} from '@sz/constants';

export function HowToShootStepFour() {
  return (
    <View>
      <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
        {`Face-On/Hand-Held Instructions\n`}
      </Text>
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`Have your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. \n`}
      </Text>
      <FastImage
        style={[tw`w-full border-4 border-Neutral-White`, { aspectRatio: HOW_TO_SHOOT_GIF_ASPECT_RATIO }]}
        source={{ cache: FastImage.cacheControl.immutable, priority: FastImage.priority.high, uri: FaceOnGIF }}
      />
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`\nHave your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. \n`}
      </Text>
      <FastImage
        style={[tw`w-full border-4 border-Neutral-White`, { aspectRatio: HOW_TO_SHOOT_GIF_ASPECT_RATIO }]}
        source={{ cache: FastImage.cacheControl.immutable, priority: FastImage.priority.high, uri: LandscapeGIF }}
      />
    </View>
  );
}
