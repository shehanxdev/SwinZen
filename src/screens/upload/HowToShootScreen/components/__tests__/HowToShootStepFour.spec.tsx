import { render } from '@testing-library/react-native';
import * as React from 'react';

import { HowToShootStepFour } from '../HowToShootStepFour';

describe('How to shoot step four', () => {
  const getRenderedScreen = () => render(<HowToShootStepFour />);

  it(`Should render How to shoot step four correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
