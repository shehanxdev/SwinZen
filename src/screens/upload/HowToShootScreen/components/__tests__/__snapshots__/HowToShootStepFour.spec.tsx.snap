// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`How to shoot step four Should render How to shoot step four correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#C0DBBF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "auto",
        },
      ]
    }
  >
    Face-On/Hand-Held Instructions

  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    Have your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. 

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "borderColor": "#FFFFFF",
            "borderWidth": 4,
            "width": "100%",
          },
          {
            "aspectRatio": 1.7378640776699028,
          },
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://files.dev.swingzen.com/gifs/faceon.gif",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    
Have your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. 

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "borderColor": "#FFFFFF",
            "borderWidth": 4,
            "width": "100%",
          },
          {
            "aspectRatio": 1.7378640776699028,
          },
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://files.dev.swingzen.com/gifs/Landscape.gif",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
